AutoFinance.Broker/InteractiveBrokers/Controllers/TwsControllerBase.cs

```diff
@@ -696,58 +834,148 @@ public Task<List<PositionStatusEventArgs>> RequestPositions(CancellationToken ca
         {
             // Set the operation to cancel after 10 seconds
             CancellationTokenSource tokenSource = new CancellationTokenSource(10000);
-            return this.RequestSecurityDefinitionOptionParameters(underlyingSymbol, exchange, underlyingSecType, underlyingConId, tokenSource.Token);
+            return RequestSecurityDefinitionOptionParameters(
+                underlyingSymbol,
+                exchange,
+                underlyingSecType,
+                underlyingConId,
+                this.twsRequestIdGenerator,
+                this.twsCallbackHandler,
+                this.clientSocket,
+                tokenSource.Token);
         }
 
         /// <summary>
-        /// Request security definition parameters.
-        /// This is mainly used for finding strikes, multipliers, exchanges, and expirations for options contracts.
+        /// Set the type for the market data feed
         /// </summary>
-        /// <param name="underlyingSymbol">The underlying symbol</param>
-        /// <param name="exchange">The exchange</param>
-        /// <param name="underlyingSecType">The underlying security type</param>
-        /// <param name="underlyingConId">The underlying contract ID, retrieved from the Contract Details call</param>
+        /// <param name="marketDataTypeId">The feed level</param>
+        public void RequestMarketDataType(int marketDataTypeId)
+        {
+            this.clientSocket.RequestMarketDataType(marketDataTypeId);
+        }
+
+        /// <summary>
+        /// Get the PnL of the account.
+        /// </summary>
+        /// <param name="accountCode">The account code</param>
+        /// <param name="modelCode">The model code</param>
+        /// <returns>The PnL account update event from TWS.</returns>
+        public Task<PnLEventArgs> RequestPnL(
+            string accountCode,
+            string modelCode)
+        {
+            // Set the operation to cancel after 5 seconds
+            CancellationTokenSource tokenSource = new CancellationTokenSource(5000);
+            return this.RequestPnL(accountCode, modelCode, tokenSource.Token);
```

Comment:
`RequestPnl` is a subscription method, so there's no need to cancel with a token. `RequestPnlSingle`, however, returns the PNL without subscribing, so should listen for the callback (with a cancellation token), and return when the event is fired.

Reply:
Ok.